components/BottomAppBar/src/private/MDCBottomAppBarView.m

```diff
@@ -0,0 +1,250 @@
+    pathAnimation.fromValue = (id)self.bottomBarLayer.presentationLayer.path;
+    pathAnimation.toValue = (__bridge id _Nullable)(cutPath);
+    pathAnimation.fillMode = kCAFillModeForwards;
+    pathAnimation.removedOnCompletion = NO;
+    pathAnimation.delegate = self;
+    [pathAnimation setValue:kMDCBottomAppBarViewPathString
+                     forKey:kMDCBottomAppBarViewAnimKeyString];
+    [self.bottomBarLayer addAnimation:pathAnimation forKey:kMDCBottomAppBarViewPathString];
+  } else {
+    self.bottomBarLayer.path = cutPath;
+  }
+}
+
+- (void)healBottomAppBarView:(BOOL)animated  {
+  CGPathRef withoutCutPath =
+      [self.bottomBarLayer pathWithoutCutFromRect:self.bounds
+                           floatingButtonPosition:self.floatingButtonPosition
+                                  layoutDirection:self.layoutDirection];
+  if (animated) {
+    CABasicAnimation *pathAnimation =
+        [CABasicAnimation animationWithKeyPath:kMDCBottomAppBarViewPathString];
+    pathAnimation.duration = kMDCFloatingButtonEnterDuration;
+    pathAnimation.fromValue = (id)self.bottomBarLayer.presentationLayer.path;
+    pathAnimation.toValue = (__bridge id _Nullable)(withoutCutPath);
+    pathAnimation.fillMode = kCAFillModeForwards;
+    pathAnimation.removedOnCompletion = NO;
+    pathAnimation.delegate = self;
+    [pathAnimation setValue:kMDCBottomAppBarViewPathString
+                     forKey:kMDCBottomAppBarViewAnimKeyString];
+    [self.bottomBarLayer addAnimation:pathAnimation forKey:kMDCBottomAppBarViewPathString];
+  } else {
+    self.bottomBarLayer.path = withoutCutPath;
+  }
+}
+
+- (void)animationDidStop:(CAAnimation *)anim finished:(BOOL)flag {
+  if (flag) {
+    [self renderPathBasedOnFloatingButtonVisibility:NO];
+    NSString *animKeyString = [anim valueForKey:kMDCBottomAppBarViewAnimKeyString];
+    if ([animKeyString isEqualToString:kMDCBottomAppBarViewPathString]) {
+      [self.bottomBarLayer removeAllAnimations];
+    } else if ([animKeyString isEqualToString:kMDCBottomAppBarViewPositionString]) {
+      [self.floatingButton.layer removeAllAnimations];
+    }
+  }
+}
+
+- (void)setFloatingButtonVisible:(BOOL)floatingButtonVisible {
+  [self setFloatingButtonVisible:floatingButtonVisible animated:NO];
+}
+
+- (void)setFloatingButtonVisible:(BOOL)floatingButtonVisible animated:(BOOL)animated {
+  if (_floatingButtonVisible == floatingButtonVisible) {
+    return;
+  }
+  _floatingButtonVisible = floatingButtonVisible;
+  if (floatingButtonVisible) {
+    self.floatingButton.hidden = NO;
+    [self cutBottomAppBarView:animated];
+    [self.floatingButton expand:animated completion:^{
+      self.floatingButton.hidden = !self.floatingButtonVisible;
+    }];
+  } else {
+    [self healBottomAppBarView:animated];
+    [self.floatingButton collapse:animated completion:^{
+      self.floatingButton.hidden = !self.floatingButtonVisible;
+    }];
+  }
+}
+
+- (void)moveFloatingButtonCenter:(BOOL)animated {
```

Comment:
Per UIKIt, add Animated: to method name.

Reply:
Done.